File: frontend/src/api/interface/host-tool.ts
export namespace HostTool {
    export interface HostTool {
        type: string;
        config: {};
    }

    export interface Supervisor extends HostTool {
        configPath: string;
        includeDir: string;
        logPath: string;
        isExist: boolean;
        init: boolean;
        msg: string;
        version: string;
        status: string;
        ctlExist: boolean;
        serviceName: string;
    }

    export interface SupervisorConfigUpdate {
        content?: string;
    }

    export interface SupervisorConfigRes {
        type: string;
        content: string;
    }

    export interface SupervisorInit {
        type: string;
        configPath: string;
        serviceName: string;
    }

    export interface SupervisorProcess {
        operate: string;
        name: string;
        command: string;
        user: string;
        dir: string;
        numprocs: string;
        status?: ProcessStatus[];
        autoRestart: string;
        autoStart: string;
        environment: string;
    }

    export interface ProcessStatus {
        PID: string;
        status: string;
        uptime: string;
        name: string;
    }

    export interface ProcessReq {
        operate: string;
        name: string;
    }

    export interface ProcessFileGetReq {
        name: string;
        file: string;
    }

    export interface ProcessFileReq {
        operate: string;
        name: string;
        content?: string;
        file: string;
    }
}

File: agent/app/dto/response/host_tool.go
package response

type HostToolRes struct {
	Type   string      `json:"type"`
	Config interface{} `json:"config"`
}

type Supervisor struct {
	ConfigPath  string `json:"configPath"`
	IncludeDir  string `json:"includeDir"`
	LogPath     string `json:"logPath"`
	IsExist     bool   `json:"isExist"`
	Init        bool   `json:"init"`
	Msg         string `json:"msg"`
	Version     string `json:"version"`
	Status      string `json:"status"`
	CtlExist    bool   `json:"ctlExist"`
	ServiceName string `json:"serviceName"`
}

type HostToolConfig struct {
	Content string `json:"content"`
}

type SupervisorProcessConfig struct {
	Name     string          `json:"name"`
	Command  string          `json:"command"`
	User     string          `json:"user"`
	Dir      string          `json:"dir"`
	Numprocs string          `json:"numprocs"`
	Msg      string          `json:"msg"`
	Status   []ProcessStatus `json:"status"`
}

type ProcessStatus struct {
	Name   string `json:"name"`
	Status string `json:"status"`
	PID    string `json:"PID"`
	Uptime string `json:"uptime"`
	Msg    string `json:"msg"`
}
